features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/components/APIDetailsTopMenu.jsx

```diff
@@ -59,13 +66,51 @@ const styles = (theme) => ({
         textOverflow: 'ellipsis',
         whiteSpace: 'nowrap',
     },
+    downloadApi: {
+        display: 'flex',
+        flexDirection: 'column',
+        textAlign: 'center',
+        justifyContent: 'center',
+        cursor: 'pointer',
+        height: 70,
+        color: theme.palette.getContrastText(theme.palette.background.paper),
+    },
+    downloadApiFlex: {
+        display: 'flex',
+        flexDirection: 'column',
+    },
 });
 
+
 const APIDetailsTopMenu = (props) => {
     const {
         classes, theme, api, isAPIProduct, imageUpdate,
     } = props;
     const isVisibleInStore = ['PROTOTYPED', 'PUBLISHED'].includes(api.lifeCycleStatus);
+    /**
+ * The component for advanced endpoint configurations.
+ * @param {string} name The name of the
+ * @param {string} version Version of the API
+ * @param {string} provider Provider of the API
+ * @param {string} format Weather to recive files in YALM of JSON format
+ * @returns {zip} Zpi file containing the API directory.
+ */
+    function exportAPI() {
+        const restApi = new API();
+        return restApi.exportApi(api.id).then((zipFile) => {
+            return Utils.forceDownload(zipFile);
+        }).catch((error) => {
+            if (error.response) {
+                Alert.error(error.response.body.description);
+            } else {
+                const message = 'Something went wrong while downloding the API';
```

Comment:
typo: downloding -> downloading
Also, we need to use internationalization.

Check GeneralConfiguration.jsx.

```
                    Alert.error(intl.formatMessage({
                        id: 'Apis.Details.Endpoints.GeneralConfiguration.Certificates.certificate.error',
                        defaultMessage: 'Something went wrong while adding the certificate.',
                    }));
```

include intl in props,

use injectIntl like below:

export default injectIntl(withStyles(styles)(GeneralConfiguration));